scheduler/scheduler.go

```diff
@@ -75,46 +156,82 @@ func (s *Scheduler) ReserveResources(hostID string, force bool, resourceRequests
 		// If the host isn't present, it is most likely that it hasn't been registered with the scheduler yet.
 		// When it is, this reservation will get counted by the initial population.
 		logrus.Warnf("Host %v not found for reserving %v. Skipping reservation", hostID, resourceRequests)
-		return nil
+		return nil, nil
 	}
 
 	i := 0
 	var err error
-	reserveLog := bytes.NewBufferString(fmt.Sprintf("New pool amounts on host %v:", hostID))
+	data := map[string]interface{}{}
+	portsRollback := map[string]interface{}{}
+	reserveLog := bytes.NewBufferString(fmt.Sprintf("New pool amount on host %v:", hostID))
+L:
 	for _, rr := range resourceRequests {
-		p, ok := h.pools[rr.Resource]
+		p, ok := h.pools[rr.GetResourceType()]
 		if !ok {
-			logrus.Warnf("Pool %v for host %v not found for reserving %v. Skipping reservation", rr.Resource, hostID, rr)
+			logrus.Warnf("Pool %v for host %v not found for reserving %v. Skipping reservation", rr.GetResourceType(), hostID, rr)
 			continue
 		}
+		PoolType := p.GetPoolType()
+		switch PoolType {
+		case computePool:
+			pool := p.(*ComputeResourcePool)
+			request := rr.(AmountBasedResourceRequest)
+			if !force && pool.Used+request.Amount > pool.Total {
+				err = OverReserveError{hostID: hostID, resourceRequest: rr}
+				break L
+			}
 
-		if !force && p.used+rr.Amount > p.total {
-			err = OverReserveError{hostID: hostID, resourceRequest: rr}
-			break
+			pool.Used = pool.Used + request.Amount
+			i++
+			reserveLog.WriteString(fmt.Sprintf(" %v total: %v used: %v.", request.Resource, pool.Total, pool.Used))
+		case portPool:
+			pool := p.(*PortResourcePool)
+			request := rr.(PortBindingResourceRequest)
+			result, er := PortReserve(pool, request)
+			logrus.Infof("Host-UUID %v, PortPool Map %v", hostID, pool.PortBindingMap)
+			if er != nil {
+				err = er
+				portsRollback = result
+				break L
+			} else {
+				data[request.Resource] = result
```

Comment:
BUG! If I send more than one port request, I only get 1 back to program in cattle